fecgov/openFEC
# Upgrade marshmallow to 2_16_3 and marshmallow-sqlalchemy to 0_15_0

## Summary (required)
snyk **MEDIUM SEVERITY** issue: [https://app.snyk.io/vuln/SNYK-PYTHON-MARSHMALLOW-72559](https://app.snyk.io/vuln/SNYK-PYTHON-MARSHMALLOW-72559)
we need upgrade marshmallow version to solve snyk alert.

- Resolves # [_3479_]
[https://github.com/fecgov/openFEC/issues/3479](https://github.com/fecgov/openFEC/issues/3479)

## Include a summary of proposed changes.
1)upgrade marshmallow from 2.6 to 2.16.3
and marshmallow-sqlalchemy from 0.8.0 to 0.15.0
2)new marshmallow version doesn't  support string to convert datetime automatically.
so, change datatype from **datetime** --> **date** in args.py
3)create some key-value pair in docs.py to make python code neatly.
4)reformat some python codes
5)rename tests/test_operation_logs.py to tests/test_operations_log.py, make consistent with endpoint name.

## How to test the changes locally
1)checkout branch
2)`pip install -r requirements.txt`
3)`python endpoints_walk.py`
make sure pass all endpoints (Thanks Charlie create this script)
4)`pytest`
5)test all endpoints with filter by date.
ex: 
(a)tag: dates
path: /calendar-dates/

(b)tag:filings
path: /operations-log/

(c)tag: financial
path: reports/{committee_type}/

(d)tag: efilings
path: /efile/filings/


## Impacted areas of the application
all endpoints will be involved.
